### PR TITLE
fix(dbt-fal): only send relevant adapter package to isolate

### PR DESCRIPTION
**Pull request by mederka**

### CODE REVIEW

File: adapter/src/dbt/adapters/fal_experimental/adapter.py [hunk 1]
```diff
@@ -80,7 +81,11 @@ def run_in_environment_with_adapter(
             raise RuntimeError(
                 "BigQuery credential method `service-account` is not supported." + \
                 " Please use `service-account-json` instead")
-        deps = [i for i in get_default_pip_dependencies(is_remote=True) if i.startswith('dbt-')]
+        deps = [
+            i
+            for i in get_default_pip_dependencies(is_remote=True, adapter_type=adapter_type)
+            if i.startswith('dbt-')
```
Comment (chamini2) [inline adapter/src/dbt/adapters/fal_experimental/adapter.py:87]: atl: why is it that we have to filter `dbt-`

File: adapter/src/dbt/adapters/fal_experimental/teleport.py [hunk 2]
```diff
@@ -87,7 +88,11 @@ def run_in_environment_with_teleport(
     The environment_name must be defined inside fal_project.yml file
     in your project's root directory."""
     if type(environment) == IsolateServer:
-        deps = [i for i in get_default_pip_dependencies(is_teleport=True) if i.startswith('dbt-')]
+        deps = [
+            i
+            for i in get_default_pip_dependencies(adapter_type, is_teleport=True)
```
Comment (chamini2) [inline adapter/src/dbt/adapters/fal_experimental/teleport.py:93]: `is_remote=True`?

File: adapter/src/dbt/adapters/fal_experimental/utils/environments.py [hunk 3]
```diff
@@ -219,27 +219,14 @@ def _parse_remote_config(config: Dict[str, Any], parsed_config: Dict[str, Any])
...
-    # contains both the main exporter of the top-level name (dbt-core) as well as all the packages that
-    # export anything to that namespace:
-    #   {"dbt": ["dbt-core", "dbt-postgres", "dbt-athena-adapter"]}
-    #
-    # This won't only include dbt.adapters.xxx, but anything that might export anything to the dbt namespace
-    # (e.g. a hypothetical plugin that only exports stuff to dbt.includes.xxx) which in theory would allow us
-    # to replicate the exact environment.
-    package_distributions = importlib_metadata.packages_distributions()
-    for dbt_plugin_name in package_distributions.get("dbt", []):
+    dbt_adapter = f"dbt-{adapter_type}"
```
Comment (chamini2) [inline adapter/src/dbt/adapters/fal_experimental/utils/environments.py:226]: Does this work for athena and other non-standard adapter?

**Review by chamini2 — COMMENTED** [review]
Did we test it having many adapters installed?

File: adapter/src/dbt/adapters/fal_experimental/utils/environments.py (hunk 3 above)
Comment (mederka) [reply adapter/src/dbt/adapters/fal_experimental/utils/environments.py:226]: Great question! We should expand our integration tests to include athena, etc.

File: adapter/src/dbt/adapters/fal_experimental/adapter.py (hunk 1 above)
Comment (mederka) [reply adapter/src/dbt/adapters/fal_experimental/adapter.py]: Updated the code to handle this in environments.py

**Pull request merged**